modules/calib3d/src/fisheye.cpp

```diff
@@ -389,21 +392,19 @@ void cv::fisheye::undistortPoints( InputArray distorted, OutputArray undistorted
 
         double scale = 0.0;
 
-        if (fabs(theta_d) > 1e-8)
+        if (fabs(theta_d) > criteria.epsilon)
```

Comment:
"Criteria type, can be one of: COUNT, EPS or COUNT + EPS": https://docs.opencv.org/4.x/d9/d5d/classcv_1_1TermCriteria.html#a56fecdc291ccaba8aad27d67ccf72c57
So need extra condition, if `criteria.epsilon` is valid value, but not random noise.